### DISCUSSION TITLE
Is it possible to implement the functionality of this script using resticprofile?

### DISCUSSION THREAD

**Discussion by 2q2code** [creativeprojects/resticprofile#430] [Q&A] [answered]
I love resticprofile, and ideally would like to have all my backups controlled using just one mechanism. Here is a script I had to write because I couldn't figure out how to back up postgresql databases individually, dynamically, using resticprofile alone. The reason I break up the database backup files is that I have worked with them on restore before and a single file is almost completely unmanageable, as it's far too large to view/edit. I may in future break this script up further into schema and data.

``` zsh
#!/usr/bin/zsh

if ! [[ $USER = "restic" ]]; then
  echo Must be run as user 'restic'
  exit 1
fi

# dump all databases into seperate files using the postgresql custom format
# that can be written by pg_dump and read by pg_restore - to stdin and to restic

EXT="pgdump" # pgdump custom extension (own)
PGDUMP=$(command -v pg_dump)
PGDUMPALL=$(command -v pg_dumpall)
PSQL=$(command -v psql)
RESTIC=$(command -v restic)
GET_PASSWORD="/usr/local/lib/resticprofile/get_restic_password.zsh"
COMPRESSION=none
REPO_BASE="rest:http://restic.example.com/postgresql_"
DB_REPO_GLOBALS=${REPO_BASE}globals
DB_REPO_DATABASES=${REPO_BASE}databases

export RESTIC_REST_USERNAME=admin
export RESTIC_REST_PASSWORD=admin

# create a directory called dump in the working directory

echo "Backing up globals..."
export RESTIC_REPOSITORY=$DB_REPO_GLOBALS
export RESTIC_PASSWORD=$(${GET_PASSWORD} postgresql_globals)
$PGDUMPALL --globals-only | $RESTIC backup --stdin-filename "globals.sql" --stdin --verbose

# get a list of all databases from psql
echo "Backing up databases..."
DBLIST=$($PSQL -l | command sed -E -e '/^(  |-|\().+$/d' -e '/^$/d' -e '/^\s+template(0|1).+$/d'  | command awk '{ print $1; }')

export RESTIC_REPOSITORY=$DB_REPO_DATABASES
export RESTIC_PASSWORD=$(${GET_PASSWORD} postgresql_databases)

while IFS=$'\n' read -n DB; do
  echo "Backing up '$DB'...";
  $PGDUMP -d "$DB" --format=custom --compress=$COMPRESSION | $RESTIC backup --stdin-filename "$DB.dump" --stdin --verbose
  echo $DB
done <<< $DBLIST

echo "Done"

```
N.B. Compression is set to _none_ because it allows restic to handle deduplication and compression most effectively, and also reduces load on the server (win/win!).

**Comment by creativeprojects** [comment, marked as answer]
Roughly yes. You need a profile per DB dump.
You can use a trick to make dynamic configuration profiles, explained in this discussion:

https://github.com/creativeprojects/resticprofile/discussions/414#discussioncomment-10870818

I think I should put it in the documentation 👍🏻

**Reply by 2q2code** [reply]
This is exactly what I was looking for and I did SCAN the ever loving CRAP out of the docs first so yeah probably a good call there. Also, for the bleeding edge desire in all of us, will v2 config syntax be seeing the light of approved usage day any time soon?

**Reply by creativeprojects** [reply]
The group scheduling landed in v0.29.0: it was the last big chunk of work for `v2`.
Now it just needs a bit of polishing and that's going to be ready (finally!)

**Reply by 2q2code** [reply]
Yay!!! 🥳
